zerver/lib/actions.py

```diff
@@ -2883,7 +2883,9 @@ def notify_realm_emoji(realm):
     send_event(event, user_ids)
 
 def do_add_realm_emoji(realm, name, img_url):
-    RealmEmoji(realm=realm, name=name, img_url=img_url).save()
+    emoji = RealmEmoji(realm=realm, name=name, img_url=img_url)
+    emoji.full_clean()
```

Comment:
`full_clean` is needed to run validators on the model field. Forms call it for example, but `save()` method does not. Here's an old thread that is still valid: https://groups.google.com/forum/#!topic/django-developers/uIhzSwWHj4c